src/vmm/src/cpu_config/x86_64/custom_cpu_template.rs

```diff
@@ -27,12 +29,18 @@ impl GetCpuTemplate for Option<CpuTemplateType> {
                 CpuTemplateType::Custom(template) => Ok(Cow::Borrowed(template)),
                 CpuTemplateType::Static(template) => {
                     let vendor_id = get_vendor_id_from_host().map_err(GetCpuVendor)?;
+                    let cpu_model = CpuModel::get_cpu_model();
                     match template {
                         StaticCpuTemplate::C3 => {
                             if &vendor_id != VENDOR_ID_INTEL {
                                 return Err(CpuVendorMismatched);
```

Comment:
This is not really needed anymore - all of the 3 cpu models below are Intel

goes for every branch actually

Reply:
isn't the vendor check required to error AMD?

Reply:
We would reject any AMD CPU in the check that verifies the exact CPU model anyway, because the only 3 CPU models we allow are Intel ones.

Reply:
Although I don't think FMS value conflicts between vendors, I'd like to keep it just in case.

Reply:
Ah, didn't realize we dont have guarantees about these not being reused. Sorry!

In that case, let's also hoist this check above the match though? E.g. have some sort of `StaticCpuTemplate::supported_vendor(&self) -> vendor_t` and just check that once against the vendor_id, instead of doing it in every branch

Reply:
yeah, makes sense to me!